src/manifest.rs

```diff
@@ -302,39 +284,37 @@ impl Manifest {
     ///
     /// ```
     /// # extern crate cargo_edit;
-    /// # extern crate toml;
+    /// # extern crate toml_edit;
     /// # fn main() {
     ///     use cargo_edit::{Dependency, Manifest};
-    ///     use toml;
+    ///     use toml_edit;
     ///
-    ///     let mut manifest = Manifest { data: toml::value::Table::new() };
+    ///     let mut manifest = Manifest { data: toml_edit::Document::new() };
     ///     let dep = Dependency::new("cargo-edit").set_version("0.1.0");
     ///     let _ = manifest.insert_into_table(&vec!["dependencies".to_owned()], &dep);
     ///     assert!(manifest.remove_from_table("dependencies", &dep.name).is_ok());
     ///     assert!(manifest.remove_from_table("dependencies", &dep.name).is_err());
-    ///     assert!(manifest.data.is_empty());
+    ///     assert!(manifest.data["dependencies"].is_none());
     /// # }
     /// ```
     pub fn remove_from_table(&mut self, table: &str, name: &str) -> Result<()> {
-        let manifest = &mut self.data;
-        let entry = manifest.entry(String::from(table));
-
-        match entry {
-            Entry::Vacant(_) => Err(ErrorKind::NonExistentTable(table.into())),
-            Entry::Occupied(mut section) => {
-                let result = match *section.get_mut() {
-                    toml::Value::Table(ref mut deps) => deps.remove(name)
-                        .map(|_| ())
-                        .ok_or_else(|| ErrorKind::NonExistentDependency(name.into(), table.into())),
-                    _ => Err(ErrorKind::NonExistentTable(table.into())),
-                };
-                if section.get().as_table().map(|x| x.is_empty()) == Some(true) {
-                    section.remove();
+        if !self.data[table].is_table_like() {
+            Err(ErrorKind::NonExistentTable(table.into()))?;
+        } else {
+            {
+                let dep = &mut self.data[table][name];
+                if dep.is_none() {
+                    Err(ErrorKind::NonExistentDependency(name.into(), table.into()))?;
                 }
-                result
+                // remove the dependency
+                *dep = toml_edit::Item::None;
             }
-        }?;
 
+            // remove table if empty
+            if self.data[table].as_table_like().unwrap().is_empty() {
```

Comment:
It is not possible to implement a `remove()` method on `TableLike`, because table does not hold a reference neither to its parent, nor to the `Item` containing it .